auth/api/iam/api.go

```diff
@@ -595,6 +608,203 @@ func (r Wrapper) StatusList(ctx context.Context, request StatusListRequestObject
 	return StatusList200JSONResponse(*cred), nil
 }
 
+func (r Wrapper) RequestOid4vciCredentialIssuance(ctx context.Context, request RequestOid4vciCredentialIssuanceRequestObject) (RequestOid4vciCredentialIssuanceResponseObject, error) {
+	if request.Body == nil {
+		// why did oapi-codegen generate a pointer for the body??
+		return nil, core.InvalidInputError("missing request body")
+	}
+	// Parse and check the requester
+	requestHolder, err := did.ParseDID(request.Did)
+	if err != nil {
+		log.Logger().WithError(err).Errorf("could not resolve DID: %s", request.Did)
+		return nil, core.NotFoundError("did not found: %w", err)
+	}
+	isWallet, err := r.vdr.IsOwner(ctx, *requestHolder)
+	if err != nil {
+		log.Logger().WithError(err).Errorf("unknown DID in this node: %s", request.Did)
+		return nil, err
+	}
+	if !isWallet {
+		err := core.InvalidInputError(fmt.Sprintf("did not owned by this node: %s", request.Did))
+		log.Logger().WithError(err).Errorf("did not owned by this node: %s", request.Did)
+		return nil, err
+	}
+
+	// Parse the issuer
+	issuerDid, err := did.ParseDID(request.Body.Issuer)
+	if err != nil {
+		log.Logger().WithError(err).Errorf("could not resolve Issuer DID: %s", request.Body.Issuer)
+		return nil, core.NotFoundError("did not found: %w", err)
+	}
+	// Fetch the endpoints
+	authorizationEndpoint, _, _, err := r.tokenEndpoint(ctx, *issuerDid)
+	if err != nil {
+		log.Logger().WithError(err).Errorf("cannot locate endpoints for did: %s", issuerDid.String())
+		return nil, err
+	}
+	endpoint, err := url.Parse(*authorizationEndpoint)
+	if err != nil {
+		return nil, err
+	}
+	// Read and parse the authorization details
+	authorizationDetails := []byte("[]")
+	if len(request.Body.AuthorizationDetails) > 0 {
+		authorizationDetails, _ = json.Marshal(request.Body.AuthorizationDetails)
+		if err != nil {
```

Comment:
wrong error being checked, just remove this check since marshalling will never fail.

Reply:
Done